pkg/ddevapp/router_test.go

```diff
@@ -266,3 +269,140 @@ func TestDisableHTTP2(t *testing.T) {
+		ArchiveInternalExtractionPath: "drupal-8.9.20/",
+		FilesTarballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.files.tar.gz",
+		FilesZipballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.files.zip",
+		DBTarURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.sql.tar.gz",
+		DBZipURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.sql.zip",
+		FullSiteTarballURL:            "",
+		Dir:                           testcommon.CreateTmpDir(t.Name() + "_2"),
+		Type:                          nodeps.AppTypeDrupal,
+		Docroot:                       "",
+		Safe200URIWithExpectation:     testcommon.URIWithExpect{URI: "/README.txt", Expect: "Drupal is an open source content management platform"},
+		DynamicURI:                    testcommon.URIWithExpect{URI: "/node/2", Expect: "Vegan chocolate and nut brownies"},
+		FilesImageURI:                 "/sites/default/files/vegan-chocolate-nut-brownies.jpg",
+	}
+
+	app, err := ddevapp.NewApp(site1.Dir, false)
+	require.NoError(t, err)
+	app2, err := ddevapp.NewApp(site2.Dir, false)
+	require.NoError(t, err)
+
+	// Configure both apps to use the same target ports. Keep original configured ports for undoing the configuration later.
+	appHTTPPort, appHTTPSPort := app.RouterHTTPPort, app.RouterHTTPSPort
+	app.RouterHTTPPort, app.RouterHTTPSPort = targetHTTPPort, targetHTTPSPort
+	app2HTTPPort, app2HTTPSPort := app.RouterHTTPPort, app.RouterHTTPSPort
+	app2.RouterHTTPPort, app2.RouterHTTPSPort = targetHTTPPort, targetHTTPSPort
+
+	t.Cleanup(func() {
+		err = app.Stop(true, false)
+		assert.NoError(err)
+		err = app2.Stop(true, false)
+		assert.NoError(err)
+
+		// Undo the configuration of the ports.
+		app.RouterHTTPPort, app.RouterHTTPSPort = appHTTPPort, appHTTPSPort
+		app2.RouterHTTPPort, app2.RouterHTTPSPort = app2HTTPPort, app2HTTPSPort
+
+		err = app.WriteConfig()
+		assert.NoError(err)
+		err = app2.WriteConfig()
+		assert.NoError(err)
+		app.RemoveGlobalProjectInfo()
+		app2.RemoveGlobalProjectInfo()
+
+		// Finally reset the router configuration, so it does not interfere other tests.
+		err = ddevapp.StartDdevRouter()
+		require.NoError(t, err)
+
+		router, err := ddevapp.FindDdevRouter()
+		if router != nil && err == nil && router.State == "running" {
+			err = dockerutil.RemoveContainer(nodeps.RouterContainer)
+			assert.NoError(err)
+		}
+	})
+
+	// Occupy target router ports:
+	l0, err := net.Listen("tcp", "127.0.0.1:"+targetHTTPPort)
+	require.NoError(t, err)
+	defer l0.Close()
+	l1, err := net.Listen("tcp", "127.0.0.1:"+targetHTTPSPort)
+	require.NoError(t, err)
+	defer l1.Close()
+
+	// Find out which ephemeral ports the apps will use.
+	ephemeralHTTPPort, ok := ddevapp.FindAvailableRouterPort(ddevapp.MinEphemeralHTTPPort, ddevapp.MaxEphemeralHTTPPort)
+	assert.Exactly(true, ok)
```

Comment:
```suggestion
	require.True(t, ok)
```

Wouldn't that be easier?